idb/postgres/postgres.go

```diff
@@ -1675,15 +1685,26 @@ func (db *IndexerDb) txnsWithNext(ctx context.Context, tx *sql.Tx, tf idb.Transa
 	nextintra := uint64(nextintra32)
 	if err != nil {
 		out <- idb.TxnRow{Error: err}
-		close(out)
+		return
+	}
+	maxRoundLoaded, err := db.getMaxRoundLoaded(tx)
+	if err != nil {
+		out <- idb.TxnRow{Error: err}
+		return
+	}
+	if maxRoundLoaded < nextround {
```

Comment:
I think the condition needs to change slightly to check if an address filter is provided. Those results are returned oldest to newest so we wouldn't have to worry about the order.

Reply:
I think this problem applies to both situations.

Reply:
not sure if I'm following - does both queries are using the same transaction ? If so, it shouldn't be inconsistent.

Reply:
I'm also not sure what you mean. :)  Will is talking about two modes of reading transactions. If address filter is not given, transactions are returned from oldest to newest. With address filter, transactions are returned from newest to oldest.